### PR TITLE
Tested Go version and coverage updates

### PR DESCRIPTION
**Pull request by meatballhat**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
 sudo: false
+dist: trusty
 
 go:
-- 1.2.2
-- 1.3.3
-- 1.4
-- 1.5.4
-- 1.6.2
+- 1.x
```
Comment (jszwedko) [inline .travis.yml]: :+1: nifty

File: .travis.yml [hunk 2]
```diff
@@ -1,15 +1,26 @@
...
+- 1.5.x
+- 1.6.x
+- 1.7.x
 - master
 
-matrix:
-  allow_failures:
-  - go: master
+before_install:
+- find "${GOPATH%%:*}" -name '*.a' -delete
```
Comment (jszwedko) [inline .travis.yml]: Hmm, why is this needed?

File: README.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 [![GoDoc](https://godoc.org/github.com/urfave/negroni?status.svg)](http://godoc.org/github.com/urfave/negroni)
 [![Build Status](https://travis-ci.org/urfave/negroni.svg?branch=master)](https://travis-ci.org/urfave/negroni)
 [![codebeat](https://codebeat.co/badges/47d320b1-209e-45e8-bd99-9094bc5111e2)](https://codebeat.co/projects/github-com-urfave-negroni)
+[![codecov](https://codecov.io/gh/urfave/negroni/branch/master/graph/badge.svg)](https://codecov.io/gh/urfave/negroni)
```
Comment (jszwedko) [inline README.md]: :clap: such badges

File: .travis.yml (hunk 2 above)
Comment (meatballhat) [inline .travis.yml]: I added this while troubleshooting issues with the `cover` tool, where a version that came with the place was causing issues on go 1.4.x: https://travis-ci.org/urfave/negroni/jobs/186114218#L133-L138



I'm thinking I should get this fixed upstream so that there is no `cover` tool preinstalled... 🤔

Comment (jszwedko) [inline .travis.yml]: Aaah, I didn't realize it was pre-installed :+1:
